datafusion/physical-plan/src/sorts/sort.rs

```diff
@@ -935,11 +940,6 @@ impl SortExec {
         }
     }
 
-    pub fn with_filter(mut self, filter: Arc<DynamicFilterPhysicalExpr>) -> Self {
-        self.filter = Some(filter);
-        self
-    }
```

Comment:
This was unused and had slipped through the cracks. I can make a new PR to just remove these methods.